src/main/java/com/krishagni/openspecimen/msk2/importer/CarsImporter.java

```diff
@@ -0,0 +1,278 @@
+		
+		//
+		// Currently assuming each CP -> Event -> Has only one SR with a unique name
+		//
+		
+		for (SpecimenRequirement req : specimenRequirements) {
+			if (StringUtils.equals(req.getName(), srName)) {
+				return req;
+			}
+		}
+		
+		return null;
+	}
+	
+	private SpecimenRequirementDetail createSr(List<SpecimenRequirementDetail> sprDetails) throws Exception {
+		SpecimenRequirementDetail sprDetail = sprDetails.iterator().next();
+		ResponseEvent<SpecimenRequirementDetail> resp = cpSvc.addSpecimenRequirement(request(sprDetail));
+		resp.throwErrorIfUnsuccessful();
+		
+		return resp.getPayload();
+	}
+
+	private SpecimenRequirementDetail updateSr(List<SpecimenRequirementDetail> srDetails, SpecimenRequirementDetail srFromDb) throws Exception {
+		SpecimenRequirementDetail sprDetail = srDetails.iterator().next();
+		sprDetail.setId(srFromDb.getId());
+		ResponseEvent<SpecimenRequirementDetail> resp = cpSvc.updateSpecimenRequirement(request(sprDetail));
+		resp.throwErrorIfUnsuccessful();
+		
+		return resp.getPayload();
+	}
+	
+	////////////////////////////////
+	//
+	// Collection Protocol Event
+	//
+	///////////////////////////////
+	
+	private void checkEvent(CollectionProtocolDetail cpDetail, CollectionProtocolEventDetail event) throws Exception {
+		CollectionProtocolEventDetail eventDetail = eventsMap.get(event.getEventLabel());
+		
+		if (eventDetail == null) {
+			CollectionProtocolEvent eventFromDb = getEventFromDb(event.getCpShortTitle(), event.getEventLabel());
+			eventDetail = eventFromDb != null ? CollectionProtocolEventDetail.from(eventFromDb) : null;
+		}
+		
+		if (eventDetail != null) {
+			updateEvent(cpDetail, eventDetail, event);
+		} else {
+			eventDetail = createEvent(cpDetail, event);
+		}
+		
+		eventsMap.put(event.getEventLabel(), eventDetail);
+	}
+
+	private CollectionProtocolEvent getEventFromDb(String cpShortTitle, String eventLabel) {
+		return daoFactory.getCollectionProtocolDao().getCpeByShortTitleAndEventLabel(cpShortTitle, eventLabel);
+	}
+	
+	private CollectionProtocolEventDetail createEvent(CollectionProtocolDetail cpDetail,
+			CollectionProtocolEventDetail event) throws Exception {
+		CollectionProtocolEventDetail eventDetail = new CollectionProtocolEventDetail();
+		eventDetail.setCpShortTitle(cpDetail.getShortTitle());
+		eventDetail.setEventLabel(event.getEventLabel());
+		
+		ResponseEvent<CollectionProtocolEventDetail> resp = cpSvc.addEvent(request(event));
+		resp.throwErrorIfUnsuccessful();
+		
+		return resp.getPayload();
+	}
+
+	private void updateEvent(CollectionProtocolDetail cpDetail, CollectionProtocolEventDetail eventDetail,
+			CollectionProtocolEventDetail event) {
+		
+	}
+	
+	//////////////////////////
+	//
+	// Collection Protocol
+	//
+	/////////////////////////
+	
+	private CollectionProtocolDetail checkCp(CollectionProtocolEventDetail event, CarsDetail object) throws Exception {
```

Comment:
1. The input to this method should be just CollectionProtocolEventDetail
2. The method name should be createCpIfAbsent(CollectionProtocolDetail cp). This is along the same lines as the naming convention followed in Java collections API.